planner/core/plan.go

```diff
@@ -401,15 +412,34 @@ func (p *baseLogicalPlan) ExplainInfo() string {
 	return ""
 }
 
+type basePhysicalCost struct {
+	planCostInit bool
```

Comment:
Why do we need to cache them?

Reply:
For a plan `op1 -> op2 -> op3`, when generating this plan from bottom to top, `op1.CalCost()` can be invoked multiple times, so cache calculated costs to gain some performance benefits.